telegram_bot/telegram_bot.py

```diff
@@ -141,8 +149,10 @@ def _set_dict_val(self, d: dict, k: str, v: Any):
             curr_dict = curr_dict[part]
         curr_dict[key_parts[-1]] = v
     
-    def present_question(self, prompt, reply_markup=None):
-        if not self.last_question_message:
+    def present_question(self, prompt, reply_markup=None, in_new_message=False):
+        if not self.last_question_message or in_new_message:
+            if self.last_question_message:
+                bot.delete_message(self._chat_id, self.last_question_message.id)
```

Comment:
```suggestion
        if not self.last_question_message or in_new_message:
            # Either there was not a previous question or we are being asked to send a new message (i.e. not replace previous one)
            if self.last_question_message:
                # In order to send a new message, we delete the old one
                bot.delete_message(self._chat_id, self.last_question_message.id)
```

These `if`s are confusing without comments :)

Reply:
Note that if you make the change I suggested in the other comment, then this comment becomes irrelevant